packages/backend/src/peripherals/database/EventRepository.ts

```diff
@@ -27,19 +28,20 @@ export class EventRepository extends BaseRepository {
    * @param fromBlock exclusive
    * @param toBlock inclusive
    */
-  async getSortedInRange(
+  async getSortedBlockNumbersInRange(
     fromBlock: number,
     toBlock: number,
     chainId: ChainId,
-  ): Promise<EventRecord[]> {
+  ): Promise<number[]> {
     const knex = await this.knex()
-    const rows = await knex('events')
+    const blockNumbers = await knex('events')
       .where('chain_id', Number(chainId))
       .andWhere('block_number', '>', fromBlock)
       .andWhere('block_number', '<=', toBlock)
+      .distinct('block_number')
       .orderBy('block_number', 'asc')
 
-    return rows.map(toRecord)
+    return blockNumbers.map((row) => row.block_number)
```

Comment:
Couldn't we leave old func?

Reply:
The thing is that there will be block numbers with multiple records (when there are multiple txs in a block). This func ensures that we get distinct block numbers and it's also easy to test.